backend/api/generator.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+# Copyright 2018-2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+export TMP_OUTPUT=/tmp
+
+# Change directory\
+cd /go/src/github.com/kubeflow/pipelines
+# Delete currently generated code.
+rm -r -f backend/api/go_http_client/*
+rm -f -f backend/api/go_client/*
+
+# Generate *.pb.go (grpc api client) from *.proto .
+${PROTOCCOMPILER} -I. -Ibackend/api \
+    -I/go/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
+    -I/go/src/github.com/grpc-ecosystem/grpc-gateway/ \
+    -I/go/src/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/options/ \
+    -I/usr/include/ \
+    --plugin=protoc-gen-go=/go/bin/protoc-gen-go  \
+    --go_out=plugins=grpc:${TMP_OUTPUT} \
+    backend/api/*.proto
+# Generate *.pb.gw.go (grpc api rest client) from *.proto.
+${PROTOCCOMPILER} -I. -Ibackend/api \
+    -I/go/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
+    -I/go/src/github.com/grpc-ecosystem/grpc-gateway/ \
+    -I/go/src/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/options/ \
+    -I/usr/include/ \
+     --plugin=protoc-gen-grpc-gateway=/go/bin/protoc-gen-grpc-gateway \
+     --grpc-gateway_out=logtostderr=true:${TMP_OUTPUT} \
+     backend/api/*.proto
+# Move *.pb.go and *.gw.go to go_client folder.
+cp ${TMP_OUTPUT}/github.com/kubeflow/pipelines/backend/api/go_client/* ./backend/api/go_client
+# Generate *.swagger.json from *.proto into swagger folder.
+${PROTOCCOMPILER} -I. -Ibackend/api \
+    -I/go/src/github.com/protocolbuffers/protobuf/src \
+    -I/go/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
+    -I/go/src/github.com/grpc-ecosystem/grpc-gateway/ \
+    -I/go/src/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/options/ \
+    -I/go/src/github.com/protocolbuffers/protobuf/src/google/protobuf \
```

Comment:
We can replace this with `/usr/include/`

Reply:
Who does this work?

Reply:
IIUC /usr/include/google/protobuf has all the well-known proto types, so adding /usr/include allows imports from 'google/protobuf'. So we no longer need to clone proto buffers repo

Reply:
cool, thanks.